tests/test_formats.py

```diff
@@ -2201,6 +2205,57 @@ def test_insert_proxy_sample_oldest(self):
             ancestors_extra.ancestors_time[:][inserted], historical_sample_time + e
         )
 
+    def test_bad_input_truncate_ancestors(self):
+        sample_data, _ = self.get_example_data(10, 10, 40, use_sites_time=False)
+        ancestors = tsinfer.generate_ancestors(sample_data)
+        with pytest.raises(ValueError, match="Time bounds cannot be negative"):
+            ancestors.truncate_ancestors(-1, -2)
+        with pytest.raises(ValueError, match="Length multiplier cannot be negative"):
+            ancestors.truncate_ancestors(0.3, 0.5, -2)
+        with pytest.raises(ValueError, match="Upper bound must be >= lower bound"):
+            ancestors.truncate_ancestors(0.3, 0.1)
+        with pytest.raises(ValueError, match="greater than older ancestor"):
+            ancestors.truncate_ancestors(10, 20)
+        with pytest.raises(ValueError, match="No ancestors in time bound"):
+            ancestors.truncate_ancestors(0.555555, 0.555555)
+
+    def test_length_multiplier(self):
+        sample_data, _ = self.get_example_data(10, 10, 40, use_sites_time=False)
+        ancestors = tsinfer.generate_ancestors(sample_data)
+        with pytest.raises(ValueError, match="Maximum length of haplotype"):
+            ancestors.truncate_ancestors(0.4, 0.6, 50)
+        trunc_anc = ancestors.truncate_ancestors(0.9, 1, 1)
+        assert np.array_equal(ancestors.ancestors_length, trunc_anc.ancestors_length)
+
+    def test_ancestors_truncated_length(self):
```

Comment:
Be good to test a few more things

- Do we leave the younger haplotypes alone?
- What happens when we specify extreme intervals, like min=0, max=1? Or such that there's exactly one haplotype in set we're truncating?
- Do we still round-trip the data exactly when we have this turned on for various parameters? We should probably add another TestRoundTrip subclass somewhere, so that we inherit all the weird examples from them.